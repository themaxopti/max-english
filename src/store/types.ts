import { InitialStateRecucer } from '../modules/testModule/state/test.reucer'
import { SomeInitialState, SOME_NAME } from './some.slice'
import { AnyAction, CombinedState } from '@reduxjs/toolkit'
import { Reducer } from 'react'
import { NAME_TEST } from '../modules/testModule/state/test.reucer'

export type Reducers = {
  [SOME_NAME]: SomeInitialState
  [NAME_TEST]: InitialStateRecucer
}
