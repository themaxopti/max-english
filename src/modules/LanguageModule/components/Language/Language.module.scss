@import '/styles/variables';

.header__language {
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    column-gap: 3px;
    // flex-basis:  ;
    font-weight: 500;
}

.language__navigation {
    height: 21px;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    // row-gap: 30px;
}

.header__language__content {
    position: absolute;
    top: 30px;
    display: flex;
    font-weight: 500;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    row-gap: 5px;
    background-color: #3297FD;
    width: 50px;
    border-radius: 10px;
    padding: 0.5rem;
    // transform: translateX(-50%);
    left: -40%;

    div {
        color: white;
        cursor: pointer;
        display: flex;
        justify-content: center;
        align-items: center;
        width: 30px;
        border-radius: 4px;
    }
}

@media(max-width:$lg) {
    .language__navigation {
        position: static;
        align-items: flex-start;
        justify-content: flex-start;
    }

    .header__language__content {
        position: static;
        top: 0;
        // margin-top: -60px;
    }

    .header__language {
        // display: block;

    }
}