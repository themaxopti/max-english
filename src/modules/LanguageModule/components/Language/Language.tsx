import React from 'react'
import { motion } from 'framer-motion'
import s from './Language.module.scss'
import { useState, useCallback } from 'react'
import { useAppearence } from '../../../../utils/motion/language'
import { useMediaQuery } from 'react-responsive'

export const Language = () => {
  const [active, setActive] = useState(false)
  const [currentLanuage, setCurrentLanuage] = useState('en')
  const languages = ['ru', 'en', 'uk']
  const isBig = useMediaQuery({
    query: '(min-width: 992px)'
  })

  return (
    <>
      <motion.nav
        tabIndex={0}
        onBlur={() => {
          setActive(false)
        }}
        onClick={() => setActive(!active)}
        initial={false}
        animate={active ? 'open' : 'closed'}
        className={s.language__navigation}
      >
        <motion.div
          whileTap={{ scale: 1.2 }}
          whileHover={{ scale: 1.1 }}
          className={s.header__language}
        >
          <motion.div>{currentLanuage}</motion.div>
          <svg width='10' fill='black' height='10' viewBox='0 0 20 20'>
            <path d='M0 7 L 20 7 L 10 16' />
          </svg>
        </motion.div>

        <motion.div
          variants={{
            open: {
              clipPath: 'inset(0% 0% 0% 0% round 10px)',
              transition: {
                type: 'spring',
                bounce: 0,
                duration: 0.7,
                delayChildren: 0.3,
                staggerChildren: 0.05
              }
            },
            closed: {
              clipPath: 'inset(10% 50% 90% 50% round 10px)',
              transition: {
                type: 'spring',
                bounce: 0,
                duration: 0.3
              }
            }
          }}
          // variants={useAppearence()}
          className={s.header__language__content}
        >
          {languages.map((el, i) => (
            <motion.div onClick={() => setCurrentLanuage(el)} whileHover={{ scale: 1.3 }} key={i}>
              {el}
            </motion.div>
          ))}
        </motion.div>
      </motion.nav>
    </>
  )
}
