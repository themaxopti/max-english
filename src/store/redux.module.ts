import { SomeInitialState, SOME_NAME } from './some.slice'
import { persistReducer, persistStore } from 'redux-persist'
import {
  AnyAction,
  CombinedState,
  configureStore,
  EmptyObject,
  Reducer,
  ReducersMapObject,
  Slice,
  ThunkMiddleware
} from '@reduxjs/toolkit'
import { EnhancedStore } from '@reduxjs/toolkit/dist/configureStore'
import { Module } from '../core/coreModule'
import { RootState } from './store'
import { Reducers } from './types'
import thunk from 'redux-thunk'

export type GetStateType = Reducer<CombinedState<Reducers>, AnyAction>

export class ReduxModule<T extends Reducer<any, AnyAction> | ReducersMapObject<any, AnyAction>> {
  public store
  constructor(rootReducer: T) {
    this.store = configureStore({ reducer: rootReducer, middleware: [thunk] })
  }
}
