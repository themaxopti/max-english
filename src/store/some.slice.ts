import { createSlice } from "@reduxjs/toolkit";
import React from "react";


const slice2 = createSlice({
    name: 'ops',
    initialState: { some: 0 },
    reducers: {
    },
})

export const someSlice2 = slice2



// const slice3 = createSlice({
//     name: 'bops',
//     initialState: { some: 0, ops: '10' },
//     reducers: {
//     },
// })

// export const someSlice3 = slice3