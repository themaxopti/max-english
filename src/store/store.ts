import { languageReducer } from '../modules/LanguageModule/state/reducers/language.slice'
import { combineReducers, configureStore } from '@reduxjs/toolkit'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { ReduxModule } from './redux.module'
import { Reducers } from './types'
import { someSlice } from '../modules/testModule/state/test.reucer'
import { someSlice2 } from './some.slice'
import { themeReducer } from '../modules/ThemeModule/state/reducers/theme.slice'
import storage from 'redux-persist/lib/storage'
import { persistReducer } from 'redux-persist'
import { persistStore } from 'redux-persist'
import thunk from 'redux-thunk'

const rootReducer = combineReducers({
  // learningNavbar: learningNavbarReducer,
  theme: themeReducer,
  language: languageReducer
})

const persistConfig = {
  key: 'root',
  storage
  // blackList: ['theme']
}
const persistedReducer = persistReducer(persistConfig, rootReducer)

// export type RootReducer = typeof rootReducer
const store = configureStore({ reducer: persistedReducer, middleware: [thunk] })

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const persistor = persistStore(store)
export default store
