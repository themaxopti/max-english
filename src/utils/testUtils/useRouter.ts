export function testRouter(useRouter: any, query: any, push: any) {
  useRouter.mockReturnValue({
    query,
    push
  })
}
