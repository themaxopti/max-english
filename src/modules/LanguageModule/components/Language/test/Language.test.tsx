import { Language } from '../Language'
import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from 'store/store'
import { createMockRouter } from 'utils/testUtils/createMockRouter'

describe('Test Language', () => {
  jest.useFakeTimers()

  test('Should open language burger', async () => {
    const languages = ['ru', 'en', 'ukk']

    const view = render(
      <Provider store={store}>
        <Language />
      </Provider>
    )

    const language = screen.getByTestId('language')
    const burger = screen.getByTestId('language__burger')
    const currentLanguage = screen.getByTestId('language__current')
    const burgerElement = screen.getAllByTestId(`languages`)

    expect(burger).toHaveStyle('clip-path: inset(10% 50% 90% 50% round 10px);')
    fireEvent.click(language)

    await waitFor(
      () => {
        expect(burger).toHaveStyle('clip-path: inset(0% 0% 0% 0% round 10px);')
        // expect(screen.getAllByTestId('circle')).
      },
      { timeout: 50000 }
    )

    // fireEvent.click(burgerElement)
    burgerElement.forEach((el, i) => {
      fireEvent.click(el)
      expect(currentLanguage).toHaveTextContent(`${el.textContent}`)
    })
    // expect(currentLanguage).toHaveTextContent(`${burgerElement.textContent}`)

    expect(view).toMatchSnapshot()
  })
})
