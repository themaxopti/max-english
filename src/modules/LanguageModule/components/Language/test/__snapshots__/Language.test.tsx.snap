// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Language Should open language burger 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="language__navigation"
        data-testid="language"
        tabindex="0"
      >
        <div
          class="header__language"
        >
          <div
            data-testid="language__current"
          >
            ua
          </div>
          <svg
            fill="black"
            height="10"
            viewBox="0 0 20 20"
            width="10"
          >
            <path
              d="M0 7 L 20 7 L 10 16"
            />
          </svg>
        </div>
        <div
          class="header__language__content"
          data-testid="language__burger"
          style="clip-path: inset(0% 0% 0% 0% round 10px);"
        >
          <div
            data-testid="languages"
          >
            en
          </div>
          <div
            data-testid="languages"
          >
            ru
          </div>
          <div
            data-testid="languages"
          >
            ua
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="language__navigation"
      data-testid="language"
      tabindex="0"
    >
      <div
        class="header__language"
      >
        <div
          data-testid="language__current"
        >
          ua
        </div>
        <svg
          fill="black"
          height="10"
          viewBox="0 0 20 20"
          width="10"
        >
          <path
            d="M0 7 L 20 7 L 10 16"
          />
        </svg>
      </div>
      <div
        class="header__language__content"
        data-testid="language__burger"
        style="clip-path: inset(0% 0% 0% 0% round 10px);"
      >
        <div
          data-testid="languages"
        >
          en
        </div>
        <div
          data-testid="languages"
        >
          ru
        </div>
        <div
          data-testid="languages"
        >
          ua
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
